Add remaining examples into sampling book

GStechschulte opened the pull request:
This draft PR resolves the following issues:

-  blackjax-devs/blackjax#447 
- blackjax-devs/blackjax#449
- blackjax-devs/blackjax#450
- blackjax-devs/blackjax#451
- blackjax-devs/blackjax#452
- blackjax-devs/blackjax#453
- blackjax-devs/blackjax#455

with the Periodic Orbital MCMC and Tempered SMC being added to the `\algorithms` section and the remaining being added to the `\models` section. 

To Do:

- [x] Update `references.bib`

GStechschulte committed:
Dispatch examples to examples/ or models/

GStechschulte committed:
Add remaining examples from Blackjax to the book

rlouf commented:
Note: you might want to start over because the warmup API changed in blackjax (and will be reflected in `blackjax-nightly` immediately). I updated the notebooks there so you shouldn't have to change anything.

GStechschulte commented:
> Note: you might want to start over because the warmup API changed in blackjax (and will be reflected in `blackjax-nightly` immediately). I updated the notebooks there so you shouldn't have to change anything.

Ahhh, my mistake and thanks for updating the notebooks. I will continue this tomorrow.

GStechschulte committed:
Refactor adaptation to return parameters and extra info

GStechschulte committed:
Add references and consistent front-matter

GStechschulte committed:
Add article references

GStechschulte commented:
The most recent commit includes the following:

- examples updated to include the new warmup API
- references added to the `references.bib` file and in-text references added in the case where examples included linked papers
- consistent front-matter for all the Jupytext `.md` files

rlouf commented:
Great! Any idea why "change_of_variable_hmc", "PeriodicOrbital" and "Pathfinder" are currently failing to build?

GStechschulte commented:
> Great! Any idea why "change_of_variable_hmc", "PeriodicOrbital" and "Pathfinder" are currently failing to build?

Good question, I am looking into it this weekend.

GStechschulte commented:
The "change_of_variable_hmc", "PeriodicOrbital" and "RegimeSwitchingModel" seem to be failing to build because the Jupyter notebook failed to execute, and since it failed to execute, the [cache key cannot be located](https://github.com/executablebooks/jupyter-cache/issues/51). Which is interesting because these examples were ported from commit [c0f9687f118794652d0303d060edc1cfe3ab29f7](https://github.com/blackjax-devs/blackjax/commit/c0f9687f118794652d0303d060edc1cfe3ab29f7) which built fine with the most recent Blackjax GitHub pages.

rlouf commented:
If you rebase on `main` the cache issues should disappear

GStechschulte commented:
Hey @rlouf, I rebased on main which resolved the caching issues; and all checks have passed. Thanks!

rlouf commented:
Awesome, thank you so much! Would you like to open a PR on blackjax to remove these examples from the documentation there?

I rebased the whole PR on `main` (it is better to rebase than merge the `main` branch to keep a linear history) and renamed the commits to fit the repo's conventions. Should be good to merge if CI is green.

rlouf committed:
Update reference

rlouf committed:
Add Arviz requirement

GStechschulte commented:
> Awesome, thank you so much! Would you like to open a PR on blackjax to remove these examples from the documentation there?
> 
> I rebased the whole PR on `main` (it is better to rebase than merge the `main` branch to keep a linear history) and renamed the commits to fit the repo's conventions. Should be good to merge if CI is green.

Yup, I can do that. Thanks for the Git tip!
